src/Effects/Chorus.cpp

```diff
@@ -131,41 +176,69 @@ void Chorus::out(const Stereo<float *> &input)
         inR = tmpc.r * (1.0f - lrcross) + tmpc.l * lrcross;
 
         //Left channel
-
-        //compute the delay in samples using linear interpolation between the lfo delays
-        float mdel =
-            (dl1 * (buffersize - i) + dl2 * i) / buffersize_f;
+        // reset output accumulator
+        output = 0.0f;
+        // increase delay line writing position and handle turnaround
         if(++dlk >= maxdelay)
             dlk = 0;
-        float tmp = dlk - mdel + maxdelay * 2.0f; //where should I get the sample from
-
-        dlhi  = (int) tmp;
-        dlhi %= maxdelay;
-
-        float dlhi2 = (dlhi - 1 + maxdelay) % maxdelay;
-        float dllo  = 1.0f + floorf(tmp) - tmp;
-        efxoutl[i] = cinterpolate(delaySample.l, maxdelay, dlhi2) * dllo
-                     + cinterpolate(delaySample.l, maxdelay,
-                                    dlhi) * (1.0f - dllo);
-        delaySample.l[dlk] = inL + efxoutl[i] * fb;
+        // linear interpolate from old to new value over length og the buffer
+        float dl = (dlHist * (buffersize - i) + dlNew * i) / buffersize_f;
+        // get sample with that delay form delay line and add to output accumulator
+        output += getSample(delaySample.l, dl, dlk);
```

Comment:
I don't see how this matches the two `cinterpolate` calls from the old code: This function calls it only once. Does your code intentionally modify the old behavior of the Chorus?

Reply:
yes. the old code did interpolate between two interpolated samples. if there is any audible difference it would be a lp filter. I don't think that is desired nor that it was intended.

Reply:
OK, I understand the intention, and the new code also looks much cleaner. However, in order to be sure, I would like @fundamental to take a look at this diff, as it might change existing sound slightly.

Reply:
We already discussed the interpolation in Discord. Resolving my comment.